test/rekt/resources/containersource/containersource_test.go

```diff
@@ -126,3 +129,57 @@ func Example_full() {
 	//         - name: POD_NAMESPACE
 	//           value: bar
 }
+
+func Example_withSink() {
+	ctx := testlog.NewContext()
+	images := map[string]string{}
+	cfg := map[string]interface{}{
+		"name":      "foo",
+		"namespace": "bar",
+	}
+
+	sinkRef := &duckv1.Destination{
+		Ref: &duckv1.KReference{
+			Kind:       "sinkkind",
+			Namespace:  "sinknamespace",
+			Name:       "sinkname",
+			APIVersion: "sinkversion",
+		},
+		URI: &apis.URL{Path: "uri/parts"},
+	}
+	containersource.WithSink(sinkRef)(cfg)
+
+	files, err := manifest.ExecuteYAML(ctx, yaml, images, cfg)
+	if err != nil {
+		panic(err)
+	}
+
+	manifest.OutputYAML(os.Stdout, files)
+	// Output:
+	// apiVersion: sources.knative.dev/v1
+	// kind: ContainerSource
+	// metadata:
+	//   name: foo
+	//   namespace: bar
+	// spec:
+	//   sink:
+	//     ref:
+	//       kind: sinkkind
+	//       namespace: sinknamespace
```

Comment:
/cc @pierDipi 
This test here is failing because the namespace in YAML is `bar` instead of `sinknamespace`. 
Whereas in the tests from your PR: https://github.com/knative/eventing/pull/6956/files#diff-3c5ec0b8f2dd9a17cfc3345346132d6f0f46a3e747ae152878db9e4d75e8362eR151, `sinknamespace` is in the output yaml

Reply:
We can change it accordingly